src/Microsoft.VisualStudio.Editor.Razor/RazorSyntaxFactsServiceExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public static class RazorSyntaxFactsServiceExtensions
+    {
+        public static bool IsTagHelperSpan(this RazorSyntaxFactsService syntaxFactsService, SourceSpan span, RazorSyntaxTree syntaxTree)
```

Comment:
Why an extension? Why not make it part of `RazorSyntaxFactsService`?

Reply:
I thought about putting it on the service directly but didn't for the reasons:
1. This method utilizes the services `GetTagHelperSpans` public implementation. Basically feels weird for a helper method to sit on a public contract.
2. It's totally different than the rest of the `RazorSyntaxFactsService` APIs.

Reply:
> This method utilizes the services GetTagHelperSpans public implementation. Basically feels weird for a helper method to sit on a public contract.

Correct me if I am wrong, but I don't think we should be concerned about this. They are both just implementation details. If someone re-implements this, they have to re-implement all the methods.

> It's totally different than the rest of the RazorSyntaxFactsService APIs.

It looks pretty similar to me but I don't have a strong opinion on this. If you think this should be an extension just for this reason, I think that's fine.

Reply:
> Correct me if I am wrong, but I don't think we should be concerned about this. They are both just implementation details. If someone re-implements this, they have to re-implement all the methods.

As an extension method that depends on the `GetTagHelperSpans` API they don't need to/can't re-implement `IsTagHelperSpan` because it acts off of the output of the `GetTagHelperSpans` method. Basically it works as long as `GetTagHelperSpans` works 😉

Reply:
Alright. That makes sense. 👍

Reply:
nit: Put `syntaxTree` before `span` in the parameters.